.github/workflows/ci.yml

```diff
@@ -413,3 +413,140 @@ jobs:
           name: Event File
           path: ${{ github.event_path }}
           retention-days: 1
+  deploy-snapshots:
```

Comment:
❓ everything from here with the `github.ref == 'refs/heads/main'` conditions is just there for alignment? as effectively only auto-merge would be relevant. (Given dependabot creates PRs against stable branches)

Or should we adjust the snapshot steps to work for stable branches? e.g. pushing a 3.0.X-SNAPSHOT docker image?

Reply:
Yeah, it's just for alignment. I think running this for branches other than main would be a separate topic and shouldn't stop us from backporting this.